powerdns/views.py

```diff
@@ -61,17 +70,149 @@ class DomainViewSet(OwnerViewSet):
 
     queryset = Domain.objects.all().select_related('owner')
     serializer_class = DomainSerializer
-    filter_fields = ('name', 'type')
+    filter_fields = ('name', 'type', 'owner',)
     permission_classes = (DomainPermission,)
 
 
+class RecordRequestsViewSet(ReadOnlyModelViewSet):
+    queryset = RecordRequest.objects.all().select_related(
+        'owner', 'domain', 'record', 'target_owner',
+    )
+    filter_fields = ('owner',)
+    serializer_class = RecordRequestSerializer
+
+
 class RecordViewSet(OwnerViewSet):
 
     queryset = Record.objects.all().select_related('owner', 'domain')
     serializer_class = RecordSerializer
-    filter_fields = ('name', 'content', 'domain')
+    filter_fields = ('name', 'content', 'domain', 'owner',)
     search_fields = filter_fields
 
+    def get_object(self):
+        obj = super().get_object()
+        if self.request.version == 'v1':
+            if (
+                self.request._request.method in ('PATCH', 'PUT') and
+                not can_edit(self.request.user, obj)
+            ):
+                raise exceptions.PermissionDenied(detail='No permission')
+            if (
+                self.request._request.method == 'DELETE' and
+                not can_delete(self.request.user, obj)
+            ):
+                raise exceptions.PermissionDenied(detail='No permission')
+        return obj
+
+    def get_serializer_class(self):
+        if self.request.version == 'v1':
+            serializer_class = RecordSerializer
+        if self.request.version == 'v2':
+            serializer_class = RecordSerializerV2
+        return serializer_class
+
+    def create(self, request, *args, **kwargs):
+        if self.request.version == 'v1':
+            return super().create(request, *args, **kwargs)
+        serializer = self.get_serializer(data=request.data)
+        serializer.is_valid(raise_exception=True)
+
+        record_request = RecordRequest(
+            domain_id=serializer.validated_data['domain'],
+            owner_id=request.user.id,
+        )
+        record_request.copy_records_data(serializer.validated_data.items())
+        record_request.save()
+
+        auto_accept = (
+            self.request.user.is_superuser or
+            serializer.validated_data['domain'].unrestricted or
+            self.request.user == serializer.validated_data['domain'].owner or
+            request.user.id in serializer.validated_data['domain'].authorisations.values_list('authorised', flat=True) # noqa
+        )
+        if auto_accept:
+            serializer.save()
+            record_request.record = serializer.instance
+            record_request.accept()
```

Comment:
`record_request.accept()` writes data from `record_request` to `record_request.record`.
Perhaps simply `record_request.state = RequestStates.ACCEPTED`
would be enough (instead of record_request.accept())